src/common/http.cc

```diff
@@ -681,6 +681,11 @@ serveFile(ResponseWriter& response, const char* fileName, const Mime::MediaType&
         throw HttpError(Code::Internal_Server_Error, "");
     }
 
+    res = ::close(fd);
```

Comment:
A better option is to probably create something like FdGuard(int &fd_) and use that as a member/in function scope to close the fd in a guaranteed manner? That way it will close irrespective of whether there was an exception or not.